iterators_task_1/student/include/impl.h

```diff
@@ -11,3 +12,17 @@
  * @warning YOU MUST USE STRING STREAMS AND STREAM ITERATORS!
  */
 std::string reverseWords(const std::string& input);
+
+class TokenString{
+public:
+    const std::string& getToken() const;
+    std::string& getToken();
+    bool isEmpty() const;
+private:
+    std::string m_data;
+};
+
+std::istream& operator>>(std::istream& stream, TokenString& token){
+    std::getline(stream, token.getToken(), ' ');
```

Comment:
space is a default separator for strings in stream, so I think you can remove this class at all and use default std::string